src/main/java/com/hyperlocal/server/ShopController.java

```diff
@@ -92,16 +87,14 @@ public ShopController(PubSubTemplate pubSubTemplate) {
   @GetMapping("/api/shop/{shopID}")
   public CompletableFuture<HashMap<String, Object>> getShopDetails(@PathVariable Long shopID) {
 
-    // Container obj for shop details
     HashMap<String, Object> shopDetailsMap = new HashMap<String, Object>();
 
-    // Promise: returns shop details
     CompletableFuture<HashMap<String, Object>> shopDetailsPromise = connection
         // Get Shop details:
       .sendPreparedStatement(SELECT_SHOP_STATEMENT, Arrays.asList(shopID))
       .thenCompose((QueryResult shopQueryResult) -> {
         ResultSet wrappedShopRecord = shopQueryResult.getRows();
-        if(wrappedShopRecord.size() == 0) throw new NullPointerException("Shop not found."); // No shop with supplied ShopID found
+        if(wrappedShopRecord.size() == 0) throw new RuntimeException("Not found"); // No shop with supplied ShopID found
```

Comment:
throw new ResponseStatusException(HttpStatus.NOT_FOUND, ex.getMessage(), ex);

Can't we do something like this ??